Documentation/bpf.rst

```diff
@@ -1119,10 +1119,10 @@ following commands can be used:
 
 ::
 
-    $ git clone https://git.llvm.org/git/llvm.git
-    $ cd llvm/tools
-    $ git clone --depth 1 https://git.llvm.org/git/clang.git
-    $ cd ..; mkdir build; cd build
+    $ git clone https://github.com/llvm/llvm-project.git
+    $ mv llvm-project/clang llvm-project/llvm/tools/
```

Comment:
Thanks for the patch!

Is the `mv` necessary here? My concern is that by moving the directory, we break the structure and so I subsequent `git update` or similar would not work.

Is there a reason why we could not use the instructions from https://clang.llvm.org/get_started.html?

```
cd llvm-project
mkdir build (in-tree build is not supported)
cd build
cmake -DLLVM_ENABLE_PROJECTS=clang -G "Unix Makefiles" ../llvm
make
```

Reply:
Hello @kkourt 
The only reason is because inside the llvm project they are all under the same working directory e.g. :
```
llvm-project/clang
llvm-project/llvm
..
..
```

And the compiling process seems to require clang inside llvm/tools code dir in order to complete successfuly the cmake step.
I totaly agree with you that on a dev point of view the git repos will have changes and if someone is working on the main repo (cilium - after compiling from scratch llvm) might need to revert those build/mv folders once the compiled code is installed but i think you already have those file changes because of running nested git clones  (git clone --depth 1 https://git.llvm.org/git/clang.git) will also trigger file changes to the .git of the main repo


I am more ops than dev so if i am wrong let me know (maybe its possible to find a way to force git submodules so they main repo doesn't have any file change after the build)


Thank you for your review!

Reply:
I believe, adapting the instructions as to the following:
```
git clone https://github.com/llvm/llvm-project.git
cd llvm-project
mkdir build
cd build
cmake -DLLVM_ENABLE_PROJECTS=clang -DLLVM_TARGETS_TO_BUILD="BPF;X86" -DBUILD_SHARED_LIBS=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_BUILD_RUNTIME=OFF  -G "Unix Makefiles" ../llvm
make -j $(getconf _NPROCESSORS_ONLN)
```


Would work as well. Or, am I missing something?

Reply:
Re-tested now the procedure and works fine with the original llvm-project instructions, sorry for the confusion
I re-pushed the correct instructions

Reply:
Great! Thanks!